### ISSUE TITLE
Can open API works with pathlib.Path?

### ISSUE THREAD

**Issue by min-xu-ai** [mgzip/mgzip#22]
```
from pathlib import Path
import pgzip
p = Path("file")
pgzip.open(p, "wb")
```

The above won't work but use `str(p)` works. Can this be supported by pgzip? Thanks!

**Comment by alexitx** [comment]
Addressed in 4a204a395a2b7331aefbb7bac97ae82aa13f2a95

**Comment by timhughes** [comment]
Released in 0.3.2
